app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -47,65 +47,80 @@ import {
 } from '../../common/analytics/analytics-events';
 import { AnalyticsClient } from '../../common/analytics/analytics-client';
 import { ParamTypes } from '../../common/types';
+import { CopiedToClipboard } from '../CopiedToClipboard/CopiedToClipboard';
 
 const analyticsClient = new AnalyticsClient();
 
 type CopyProps = {
 	handleCopyToClipboard: (copyRef: React.RefObject<HTMLDivElement>, elementName?: string) => Promise<void> | void;
+	setCopyButtonStateToTrue: (copyButton: CopyButtonNameEnum) => void;
+	primaryButtonName: CopyButtonNameEnum;
+	copyButtonName: CopyButtonNameEnum;
 	testId?: string
 };
 
 const CopyButton = ({
 	handleCopyToClipboard,
 	copyRef,
+	setCopyButtonStateToTrue,
+	primaryButtonName,
+	copyButtonName,
 	testId
 }: CopyProps & { copyRef: React.RefObject<HTMLDivElement> }): JSX.Element => {
 	const [isCopied, setIsCopied] = useState(false);
+	const [showIsCopiedClipBoard, setShowIsCopiedClipBoard] = useState(false);
 
 	useEffect(() => {
 		let timeoutId: NodeJS.Timeout;
-		if (isCopied) {
+		if (showIsCopiedClipBoard) {
 			timeoutId = setTimeout(() => {
-				setIsCopied(false);
+				setShowIsCopiedClipBoard(false);
 			}, 2000);
 		}
 		return () => {
 			if (timeoutId) {
 				clearTimeout(timeoutId);
 			}
 		};
-	}, [isCopied]);
+	}, [showIsCopiedClipBoard]);
 
 	return (
 		<div className={cx(jenkinsSetupCopyButtonContainer)} >
 			<Button
-				iconBefore={<CopyIcon label="Copy" size="medium" />}
+				iconBefore={isCopied ? <CheckIcon label="Copy" size="medium" /> : <CopyIcon label="Copy" size="medium" />}
 				onClick={() => {
 					if (copyRef && copyRef.current) {
 						handleCopyToClipboard(copyRef, testId);
 						setIsCopied(true);
```

Comment:
```
setIsCopied(true);
setShowIsCopiedClipBoard(true);
setCopyButtonStateToTrue(copyButtonName);
```

It seems that all of these variables are tracking similar things. We could probably consolidate this

Reply:
Although they are triggered by the same action, they are liable to different things. isCopied is responsible for controlling the icon and color of the button. showIsCopiedClipBoard control the clip board to be shown after clicked, which would disappear after 2 seconds. copyButtonStateToTrue set the state in its parent component. Happy to chat on your thoughts.

Reply:
Oh sure that makes sense. Regarding the `showIsCopiedClipboard`, it would make more sense to include the word tooltip in there to differentiate it from the other functions. I would update this to `showIsCopiedToClipboardTooltip`

Ideally, the `CopyButton` component wouldn't live in this file as it should be reusable. We should move it out to a `/common` folder at a later date, which would also make this file less confusing.